addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -1417,10 +1417,6 @@ const ColorpickerUserValueWidget = SelectUserValueWidget.extend({
         const _super = this._super.bind(this);
         const args = arguments;
 
-        if (!this.options.dataAttributes.lazyPalette === 'true') {
-            await this._renderColorPalette();
-        }
```

Comment:
This condition was wrongly changed by NBY in the problematic commit... why can't it be just restored to first render the colorpalette in a fragment if needed? I'd like to avoid the use of `getParent()` and having to implement this getColorNames function

Reply:
@bvr-odoo @robinlej I forgot about this. I still cannot merge this PR like that: if we don't have a solution, TODO comments should explain what's going on so we can investigate later. I see there is still this code right now:
```javascript
onSetColorNames: (colorNames) => {
    this.colorPaletteColorNames = colorNames
},
```
So this seems like a duplicated code to retrieves the color names, right? :thinking: (is this even needed since your hack with `getParent` and the new function? (if your function is supposed to be the new way to retrieve colors, do we still need that old one? (which is also strange and ugly somehow ^^)) :thinking: 

Side comment: the author of both remaining commits is BVR, should you be a co-author @robinlej ? Was not sure who to ping on this :)

Reply:
@qsm-odoo Not sure what type of comments you're expecting, I thought the one I wrote last time was enough. ^^' I've reworded it a bit and added a "TODO" to let it stand out, let me know if you think there should still be more to it / other comments.

I removed the `onSetColorNames` prop inside the snippets.options (it should work indeed, and as far as I can tell after testing, it does), but left it in general for the ColorPalette component (it's used in the web_editor code for more specific uses, and I guess this should anyway be the way to go once the options are truly converted to OWL).

This all being said, there's one test failing since your rebase 3 weeks ago in mass_mailing... It does work on local, so I'm investigating.

(I'm indeed author of this commit. I've added the line.)

Reply:
> Not sure what type of comments you're expecting

Basically just adding "TODO" in the comment and not having multiple ways the colors are fetched is what I expected, so what you did :wink: 

> (I'm indeed author of this commit. I've added the line.)

If you are the only one, you can also change the author entirely (`--author=`)

> I'm investigating.

Ping me once ok, thanks :+1: